tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -2577,6 +2566,197 @@ llvm::Optional<Value> convertReduceMeanOp(
   return val;
 }
 
+// Lowers ResizeBilinear and ResizeNearestNeighbor to TOSA resize.
+llvm::Optional<Value> convertResizeOp(PatternRewriter& rewriter, Operation* op,
+                                      RankedTensorType output_type,
+                                      Value input_value, StringRef mode,
+                                      bool align_corners,
+                                      bool half_pixel_centers) {
+  RankedTensorType input_type =
+      input_value.getType().dyn_cast<RankedTensorType>();
+  if (!input_type) return llvm::None;
+
+  auto input_shape = input_type.getShape();
+  auto output_shape = output_type.getShape();
+
+  size_t input_height = input_shape[1];
+  size_t input_width = input_shape[2];
+  size_t output_height = output_shape[1];
+  size_t output_width = output_shape[2];
+
+  bool input_is_qtype =
```

Comment:
nit: move these checks to the top, keeping the height/width extraction with the calculations that use them.

Reply:
Thanks. Will update in next round.